### PR TITLE
ENT-2695 Restore async logging

### PR DESCRIPTION
**Pull request by rick-r3**
Replace error code generation technique with custom event re-writer. Switch to RandomAccessFile appenders because they supposedly give higher throughput.

Also added a small unit test for async logging.

### CODE REVIEW

File: testing/test-common/src/main/resources/log4j2-test.xml [hunk 1]
```diff
@@ -1,5 +1,5 @@
 <?xml version="1.0" encoding="UTF-8"?>
-<Configuration status="info">
+<Configuration status="info" packages="net.corda.cliutils">
```
Comment (sollecitom) [inline testing/test-common/src/main/resources/log4j2-test.xml]: Nice :) out of curiosity, why do we need to different `log4j2.xml` files and, in particular, one for the driver?

File: tools/cliutils/src/main/kotlin/net/corda/cliutils/ErrorCodeRewritePolicy.kt [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+import org.apache.logging.log4j.core.config.plugins.PluginFactory
+import org.apache.logging.log4j.core.impl.Log4jLogEvent
+
+@Plugin(name = "ErrorCodeRewritePolicy", category = Core.CATEGORY_NAME, elementType = "rewritePolicy", printObject = false)
+class ErrorCodeRewritePolicy : RewritePolicy {
+
+    override fun rewrite(source: LogEvent?): LogEvent? {
+        val newMessage = source?.message?.withErrorCodeFor(source.thrown, source.level)
+        if (newMessage == source?.message) {
+            return source
```
Comment (sollecitom) [inline tools/cliutils/src/main/kotlin/net/corda/cliutils/ErrorCodeRewritePolicy.kt]: You can lift the `return` outside of the `if` block, and maybe replace it with `when`.

File: testing/test-common/src/main/resources/log4j2-test.xml (hunk 1 above)
Comment (rick-r3) [reply testing/test-common/src/main/resources/log4j2-test.xml]: Something to do with team city not working well with the other config, apparently.

File: tools/cliutils/src/main/kotlin/net/corda/cliutils/ErrorCodeRewritePolicy.kt (hunk 2 above)
Comment (rick-r3) [reply tools/cliutils/src/main/kotlin/net/corda/cliutils/ErrorCodeRewritePolicy.kt]: Will pull out return.  Will stick with `if` when it's a binary choice.

**Pull request merged**